runtimes/universal/models/gguf_language_model.py

```diff
@@ -312,6 +312,203 @@ def _generate():
                 # Regular token
                 yield item
 
+    async def chat(
```

Comment:
Looks like we created duplicate methods called `chat` and `chat_stream` on gguf model service specifically. We should re-use the existing `generate` and `generate_stream` methods to do chat completions. Their signatures can be amended as-needed to support the additional params, although the implementation may need to differ a bit in order to ensure the API is consistent across model types (transformers vs. llama cpp, etc)